src/rule/index.ts

```diff
@@ -120,6 +120,50 @@ export const removeServer = (
   return null;
 };
 
+const matchRequestBody = (
+  operationNode: a.Operation,
+  request: har.request,
+  resultOperationNode: a.Operation,
+  result: Result.Result
+) => {
+  if (!operationNode.requestBody) {
+    result.success = false;
+    resultOperationNode.requestBody = a.newRequestBodyWithError(
+      "REQUEST BODY NOT FOUND"
+    );
+    return;
+  }
+  incrementHit(operationNode.requestBody);
+  resultOperationNode.requestBody = a.newRequestBody();
+
+  if (!operationNode.requestBody.content) {
+    result.success = false;
+    resultOperationNode.requestBody.content = a.newContentBodyWithErrors(
+      "CONTENT WAS NOT FOUND"
+    );
+    return;
+  }
+  incrementHit(operationNode.requestBody.content);
+  resultOperationNode.requestBody.content = a.newContentBody();
+
+  const mimeType = request.postData?.mimeType as string;
+  if (
+    mimeType === "application/json" &&
+    !operationNode.requestBody.content[mimeType]
+  ) {
```

Comment:
Why is this dependent on `application/json`? The MIME type should always match whether we can validate the body?

Reply:
this was dues to some type wrangling, made it somewhat better in https://github.com/XenitAB/apidelta/pull/14/commits/f4fff90e8d058e32262067d1e4b9070cc2d7b3ec